packages/shared-metrics/src/gc.js

```diff
@@ -37,37 +48,46 @@ const nativeModuleLoader = require('./util/nativeModuleRetry')({
 
 let senseIntervalHandle;
 
-exports.activate = function activate() {
-  nativeModuleLoader.on('loaded', gcStats => {
-    exports.currentPayload.statsSupported = true;
-    gcStats.on('stats', stats => {
-      // gcstats exposes start and end in nanoseconds
-      const pause = (stats.end - stats.start) / 1000000;
-      gcPauseWindow.addPoint(pause);
-      const type = stats.gctype;
-      if (type === 1) {
-        minorGcWindow.addPoint(1);
-      } else if (type === 2) {
-        majorGcWindow.addPoint(1);
-      } else if (type === 4) {
-        incrementalMarkingsWindow.addPoint(1);
-      } else if (type === 8) {
-        processWeakCallbacksWindow.addPoint(1);
-      } else if (type === 15) {
-        minorGcWindow.addPoint(1);
-        majorGcWindow.addPoint(1);
-        incrementalMarkingsWindow.addPoint(1);
-        processWeakCallbacksWindow.addPoint(1);
-      }
-      exports.currentPayload.usedHeapSizeAfterGc = stats.after.usedHeapSize;
-    });
-    startSensing();
-  });
+nativeModuleLoader.on('loaded', gcStats_ => {
+  gcStats = gcStats_;
+  exports.currentPayload.statsSupported = true;
+  if (activateHasBeenCalled) {
+    actuallyActivate();
+  }
+});
+
+nativeModuleLoader.on('failed', () => {
```

Comment:
You mean remove the handlers for those events? We do not remove them. We could probably use [`once`](https://nodejs.org/api/events.html#events_emitter_once_eventname_listener) instead of `on` for that. Though I don't think it is an issue.

Reply:
Switched to `once` now because it is more appropriate for this use case.